cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java

```diff
@@ -1256,41 +1257,59 @@ public void testSearchResultPaginationWithTargetType() throws Exception {
     // reproduce the scenario that caused the issue CDAP-7881
     NamespaceId namespace = new NamespaceId("pagination_with_target_type");
     namespaceClient.create(new NamespaceMeta.Builder().setName(namespace).build());
-
     StreamId stream1 = namespace.stream("text1");
     StreamId stream2 = namespace.stream("text2");
+    DatasetId trackerDataset = namespace.dataset("_auditLog");
+    DatasetId mydataset = namespace.dataset("mydataset");
 
-    // we need this dataset to be created before the streams, so that its metadata comes before the stream's metadata
-    // in the scan in MetadataDataset#search
+    // the creation order below will determine how we see the entities in search result sorted by entity creation time
+    // in ascending order
     datasetClient.create(
-      namespace.dataset("mydataset"),
+      trackerDataset,
+      new DatasetInstanceConfiguration(Table.class.getName(), Collections.<String, String>emptyMap())
+    );
+    datasetClient.create(
+      mydataset,
       new DatasetInstanceConfiguration(Table.class.getName(), Collections.<String, String>emptyMap())
     );
 
     // create entities so system metadata is annotated
     streamClient.create(stream1);
     streamClient.create(stream2);
 
-    String sort = AbstractSystemMetadataWriter.ENTITY_NAME_KEY + " asc";
+    // do sorting with creation time here since the testSearchResultPagination does with entity name
```

Comment:
In the earlier PR where this was added there was some discrepancy in the intention and code https://github.com/caskdata/cdap/pull/7427/files#diff-b4114e6a647587816791abcd9ef15f72R1263
The comment here was itending to do test with creation time but the sorting was based on entity-name. We already have another test which was testing with entity-name so here we are doing sorting with creation-time to cover both sorting possibility.